### PR TITLE
Support shading area below lines

### PR DESCRIPTION
**Pull request by chnn**
Kapture 2019-06-12 at 11 42 14

### CODE REVIEW

File: src/utils/drawLines.ts [hunk 1]
```diff
@@ -10,24 +10,50 @@ interface DrawLinesOptions {
...
+  shadeBelow,
+  shadeBelowOpacity,
+  y0,
 }: DrawLinesOptions): void => {
   const context = canvas.getContext('2d')
 
+  if (shadeBelow) {
+    for (const {xs, ys, fill} of Object.values(lineData)) {
+      const areaGenerator = area<number>()
+        .y0(y0) // TODO
```
Comment (ebb-tide) [inline src/utils/drawLines.ts]: what is this TODO?

File: src/constants/index.ts [hunk 2]
```diff
@@ -60,6 +60,8 @@ export const LAYER_DEFAULTS: {[layerType: string]: Partial<LayerConfig>} = {
     colors: DEFAULT_COLOR_SCHEME,
     interpolation: 'linear',
     maxTooltipRows: 24,
+    shadeBelow: false,
```
Comment (ebb-tide) [inline src/constants/index.ts]: I think the masses want this to be true

File: src/components/LineLayer.tsx [hunk 3]
```diff
@@ -69,8 +71,20 @@ export const LineLayer: FunctionComponent<Props> = ({
       lineData: simplifiedLineData,
       interpolation,
       lineWidth,
+      shadeBelow,
+      shadeBelowOpacity,
+      y0: yScale(0),
```
Comment (ebb-tide) [inline src/components/LineLayer.tsx]: Should this be `y0:yScale(yDomain[0])` instead? Or do we always want to shade until zero? Also `y0` is a bit hard to understand what it's for, could it be `shadeAboveY`maybe?

File: src/utils/drawLines.ts (hunk 1 above)
Comment (chnn) [reply src/utils/drawLines.ts]: whoops, it is outdated

File: src/constants/index.ts (hunk 2 above)
Comment (chnn) [reply src/constants/index.ts]: 😈

File: src/components/LineLayer.tsx (hunk 3 above)
Comment (chnn) [reply src/components/LineLayer.tsx]: Ah yep totally, nice catch. I will rename as well.

**Pull request merged**